libvuln/updates/locks.go

```diff
@@ -2,108 +2,67 @@ package updates
 
 import (
 	"context"
-	"errors"
 	"sync"
-	"time"
-
-	"github.com/jackc/pgx/v4/pgxpool"
-
-	"github.com/quay/claircore/pkg/distlock"
-	"github.com/quay/claircore/pkg/distlock/postgres"
 )
 
-var _ LockSource = (*poolLockSource)(nil)
-
-type poolLockSource struct {
-	p   *pgxpool.Pool
-	dur time.Duration
-}
-
-func (s *poolLockSource) NewLock() distlock.Locker {
-	return postgres.NewPool(s.p, s.dur)
-}
-
-// PoolLockSource provides locks backed by Postgres advisory locks.
-//
-// The provided duration is the retry period, as documented in postgres.NewPool.
-func PoolLockSource(p *pgxpool.Pool, dur time.Duration) (LockSource, error) {
-	return &poolLockSource{
-		p: p, dur: dur,
-	}, nil
-}
-
 var _ LockSource = (*localLockSource)(nil)
-var _ distlock.Locker = (*localLockTab)(nil)
 
 type localLockSource struct {
-	sync.RWMutex
-	m map[string]chan struct{}
+	sync.Mutex
+	wait *sync.Cond
+	m    map[string]struct{}
 }
 
 // LocalLockSource provides locks backed by local concurrency primitives.
-func LocalLockSource() LockSource {
-	return &localLockSource{
-		m: make(map[string]chan struct{}),
+func LocalLockSource() *localLockSource {
```

Comment:
Maybe we could take this opportunity to update the name to `NewLocalLockSource()`